internal/core/components/rewriter.go

```diff
@@ -42,40 +43,40 @@ func (r *Rewriter) Close() {
 	}
 }
 
-func (r *Rewriter) watch(ch chan *comm.Watch) {
+func (r *Rewriter) watch(ch chan *comm.Message) {
 	for {
 		select {
 		case <-r.ctx.Done():
 			return
-		case watch := <-ch:
-			values, entities, nodes, err := watch.Decode()
+		case msg := <-ch:
+			// hand nacos notify message
+			values, entities, nodes, err := msg.Decode()
 			if err != nil {
 				log.Warnf("Rewriter decode watch message error: %s", err)
 				continue
 			}
 
+			if len(nodes) == 0 {
```

Comment:
Maybe we need to update nodes to 0 when the service is removed from the service discovery? And we need to add a warning log for it.